### ISSUE TITLE
param_shift_filter gradient estimator is biased when a branch receives zero samples

### ISSUE THREAD

**Issue by haasonsaas** [extropic-ai/torx#9]
## Severity

**High** — undermines trust in sampled training with `diff_method="param_shift_filter"`.

## Affected version

- PyPI `extro-torx==0.0.1`
- Commit `769d2f90abdfda14798fceb521143f4b99d370da` (initial public commit)

## Summary

The backward pass of `sample_expval_all_param_shift_filter` estimates a conditional expectation per branch. When no samples took a branch, it substitutes `0.0` and treats that as the true conditional expectation:

```python
# torx/psc/simulation/sampled.py
safe_counts = jnp.where(branch_counts > 0, branch_counts, 1.0)
expval_per_branch = weighted_sum / safe_counts[..., None]
expval_per_branch = jnp.where(
    branch_counts[..., None] > 0,
    expval_per_branch,
    0.0,
)
```

It then plugs that into:

```text
p_k * (E[O | k] - E[O])
```

The population identity is correct; filling empty branches with `0` makes the Monte Carlo estimator **biased**.

## Minimal counterexample

- Gate: one `PCNOT([0, 1])`
- Initial state: `[0, 1]`
- `theta = 0` (branch probability `0.5`)
- `num_samples = 1`
- Loss: `sum(expval_all)`

Because the control is `0`, both branches leave the state unchanged, so the **true derivative is exactly zero**.

With one sample:

| Branch drawn | Reported gradient |
|--------------|-------------------|
| 0            | `-0.5`            |
| 1            | `0`               |

Each occurs with probability `0.5`, so `E[estimator] = -0.25`, not zero.

Verified on the pinned commit (mean over 200 seeds ≈ `-0.27`).

Existing “unbiasedness” tests use large sample counts (e.g. 10,000), which make empty branches rare enough to hide the failure.

## Suggested fix

Use a direct score-function estimator that never divides by branch counts:

```python
score = one_hot(branch_indices, K)[..., 1:] - probs[..., 1:]
gradient = mean(observable * score)
```

For repeated shared gates, sum the score across repetitions before averaging across samples.

## Regression tests

- One-sample, branch-independent-output case above (true grad = 0).
- Assert estimator mean is near zero under many seeds with `num_samples=1`.
